src/server/utils/tracer.js

```diff
@@ -75,7 +75,11 @@ export class Tracer {
 
   // End a timer for a server phase
   endServerPhaseTimer = (serverPhasekey) => {
-    this.#serverPhaseTimers[serverPhasekey].endTimeNs = process.hrtime.bigint();
+    if (this.#serverPhaseTimers[serverPhasekey]) {
+      this.#serverPhaseTimers[serverPhasekey].endTimeNs = process.hrtime.bigint();
+    } else {
+      console.warn('Tracer Error: endServerPhaseTimer was called without startServerPhaseTimer');
+    }
```

Comment:
exit early?
```suggestion
    if (!this.#serverPhaseTimers[serverPhasekey]) {
      console.warn('Tracer Error: endServerPhaseTimer was called without startServerPhaseTimer');
      return;
    }
    this.#serverPhaseTimers[serverPhasekey].endTimeNs = process.hrtime.bigint();
```

Reply:
it's a pattern from https://effectivejs.com/ but the book is at home so I don't have a link handy to the specific chapter, sorry 😓